lang_qc/db/helper/qc.py

```diff
@@ -132,6 +133,38 @@ def qc_state_for_product_exists(
     return bool(session.execute(query).scalar_one())
 
 
+def qc_state_for_products_exists(
+    session: Session, ids: list[ChecksumSHA256], sequencing_outcomes_only: bool = False
+) -> set[ChecksumSHA256]:
+    """
+    Given a list of product IDs, returns a potentially empty subset of this list
+    as a Set object. Each product, identified by the product ID in the returned
+    set, has some QC state associated with it. If `sequencing_outcome_only`
+    boolean flag is true, the product has the `sequencing` QC state associated
```

Comment:
Suggest: "the product must have `sequencing` QC state associated with it. `library` QC is excluded".